tests/integration/ha_tests/helpers.py

```diff
@@ -185,3 +175,105 @@ async def app_name(ops_test: OpsTest) -> str:
             return app
 
     return None
+
+
+def storage_type(ops_test, app):
+    """Retrieves type of storage associated with an application.
+
+    Note: this function exists as a temporary solution until this issue is resolved:
+    https://github.com/juju/python-libjuju/issues/694
+    """
+    model_name = ops_test.model.info.name
+    proc = subprocess.check_output(f"juju storage --model={model_name}".split())
+    proc = proc.decode("utf-8")
+    for line in proc.splitlines():
+        if "Storage" in line:
+            continue
+
+        if len(line) == 0:
+            continue
+
+        if "detached" in line:
+            continue
+
+        unit_name = line.split()[0]
+        app_name = unit_name.split("/")[0]
+        if app_name == app:
+            return line.split()[3]
+
+
+def storage_id(ops_test, unit_name):
+    """Retrieves  storage id associated with provided unit.
+
+    Note: this function exists as a temporary solution until this issue is resolved:
+    https://github.com/juju/python-libjuju/issues/694
+    """
+    model_name = ops_test.model.info.name
+    proc = subprocess.check_output(f"juju storage --model={model_name}".split())
+    proc = proc.decode("utf-8")
+    for line in proc.splitlines():
+        if "Storage" in line:
+            continue
+
+        if len(line) == 0:
+            continue
+
+        if "detached" in line:
+            continue
+
```

Comment:
I understand this is a temp solution, but you could write a helper function to get_unit_storage(), return line if line==unit_name, and then index the output to get ids and storage types.

Reply:
I'll argue as above